### ISSUE TITLE
Emit error metrics

### ISSUE THREAD

**Issue by kopiczko** [kubernetes/kube-state-metrics#818]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

Due to cluster role misconfiguration kube-state-metric was unable to list resources for configured collectors.

```
kube-state-metrics-64546d9589-jpq28 kube-state-metrics E0629 10:13:54.174708       1 reflector.go:126] k8s.io/kube-state-metrics/internal/collector/builder.go:482: Failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:kube-system:kube-state-metrics" cannot list resource "deployments" in API group "apps" at the cluster scope
```

**What you expected to happen**:

kube-state-metric emits error counter ideally specific for this case but general one would also be useful.

**How to reproduce it (as minimally and precisely as possible)**:

Change group for deployments in cluster role.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): multiple 
- Kube-state-metrics image version: v1.6.0

**Comment by brancz** [comment]
I think this 100% valid. I'm not sure client-go allows extracting this today, but it should! :)

**Comment by brancz** [comment]
Closed by #866. (and released in v1.8.0)
